Spigot-Server-Patches/0648-CommandFunction-API.patch

```diff
@@ -0,0 +1,166 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Wed, 6 Jan 2021 23:38:05 -0800
+Subject: [PATCH] CommandFunction API
+
+
+diff --git a/src/main/java/io/papermc/paper/datapack/PaperCommandFunction.java b/src/main/java/io/papermc/paper/datapack/PaperCommandFunction.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..3528802b21038fc4f895204a27dc1f1de4ddbaa9
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/datapack/PaperCommandFunction.java
+@@ -0,0 +1,45 @@
++package io.papermc.paper.datapack;
++
++import net.minecraft.server.MinecraftServer;
++import org.bukkit.NamespacedKey;
++import org.bukkit.craftbukkit.util.CraftNamespacedKey;
++
++import java.util.ArrayList;
++import java.util.List;
++
++public class PaperCommandFunction implements CommandFunction {
++
++    private final net.minecraft.server.CustomFunction handle;
++    private NamespacedKey functionKey;
++    private List<String> commands;
++
++    public PaperCommandFunction(net.minecraft.server.CustomFunction handle) {
++        this.handle = handle;
++    }
++
++    @Override
++    public int execute() {
++        return MinecraftServer.getServer().getFunctionData().a(handle, MinecraftServer.getServer().getServerCommandListener().silent().withMaxPerms(2)); // Taken from /function command
++    }
++
++    @Override
++    public List<String> getCommands() {
++        if (this.commands == null) {
++            this.commands = new ArrayList<>();
++            for (net.minecraft.server.CustomFunction.c entry : handle.getEntries()) {
```

Comment:
I would love an obfhelper here but I dont think we can do that for classes, lol, screw you md5